### ISSUE TITLE
Some problem about test data.

### ISSUE THREAD

**Issue by cowry5** [graytowne/caser_pytorch#5]
```
    def fit(self, train, test, verbose=False):
        """
        The general training loop to fit the model
        Parameters
        ----------
        train: :class:`spotlight.interactions.Interactions`
            training instances, also contains test sequences
        test: :class:`spotlight.interactions.Interactions`
            only contains targets for test sequences
```

[here](https://github.com/graytowne/caser_pytorch/blob/99b89637a1fa0ca3f416dd8999b65801c01697e1/train_caser.py#L96)
, you say "test: only contains targets for test sequences". But I have check the test.txt in 'dataset/valid/test.txt'. I don’t understand the conten in this file... Shouldn't there be only three elements in each session? But I have see a lot of. 

And in `evaluate_ranking()` of `evaluate.py`, `test = test.tocsr()` ,  test is a Interactions object. I have see the src, But I don't know what that means yet. 

I'd appreciate it if I could get your help. Thank you so much!

**Comment by cowry5** [comment]
And..  I don't see any padding operation in there, did I miss something?

**Comment by graytowne** [comment]
Hi @cowry5,
Yep here test is an interaction object that contains the test.txt and the `tocsr()` method is actually making the triplets in test.txt into a sparse matrix (size: #user x #items)  and use [row.indices](https://github.com/graytowne/caser_pytorch/blob/master/evaluation.py#L76) to indicate the ground-truth items.
I use `0` for the padding, you can see [here](https://github.com/graytowne/caser_pytorch/blob/master/interactions.py#L130) for more details.

Sorry for the misleading in the code, I gonna improve the readibility once I am available.

**Comment by cowry5** [comment]
@graytowne 
Don't be sorry, Thank you so much for your reply. Your code helps me a lot!
I have know the what does tocsr() do thanks to you.
But I have saw these code in [here](https://github.com/graytowne/caser_pytorch/blob/master/interactions.py#L217)
```
        for i in range(len(tensor), 0, -step_size):
            if i - window_size >= 0:
                yield tensor[i - window_size:i]
            else:
                break
    else:
        yield tensor
```
When the tensor length is not enough, you returns it directly.
But I have tried, an error is reported when the length is insufficient.`could not broadcast input array from shape (2) into shape (5)`

 And it's okay after I did this change.
```
    if len(tensor) - window_size >= 0:
        for i in range(len(tensor), 0, -step_size):
            if i - window_size >= 0:
                yield tensor[i - window_size:i]
            else:
                break
    else:
         d = window_size - len(tensor)
         yield np.pad(tensor, (d,0), 'constant')

```

**Comment by graytowne** [comment]
Hi @cowry5, thanks for the feedbacks. You are right, I think that is a bug in my code, thanks for pointing that out. I gonna have a PR to fix it.

**Comment by graytowne** [comment]
Updates: Fixed in PR #6. Closing this issue.
